refactor: 도메인 재설계

leo0842 opened the pull request:
## 상세 내용

Close #416 

~아직 이식은 하지 않았고 해피케이스 테스트만 진행하였습니다.~

투표 도메인 재설계 및 기존 코드에 적용 완료했습니다❤️
많은 관심 바랍니다.

### 주의 사항
main, dev에 적용하기 위해서는 DB 스키마 변경이 선행되어야합니다. 
관련 내용은 [노션](https://www.notion.so/ririhan/DB-014b922cab3f411a8934d621161eba6f)에 정리해두었습니다. 

![image](https://user-images.githubusercontent.com/45311765/196631030-b45417c5-a673-44f4-b1bd-8c7e9393b292.png)

RIANAEH committed:
feat(poll): 객체지향적으로 도메인 초안 구성

RIANAEH committed:
feat(poll): JPA 살짝 적용

leo0842 committed:
Merge branch 'refactor/brand-new-poll-#416' of https://github.com/RIANAEH/2022-mo-rak into refactor/brand-new-poll-#416

RIANAEH committed:
feat(poll): JPA 꽤 적용

leo0842 committed:
Merge branch 'refactor/brand-new-poll-#416' of https://github.com/RIANAEH/2022-mo-rak into refactor/brand-new-poll-#416

leo0842 committed:
feat(poll): 익명/기명에 따른 뷰 로직 추가

leo0842 committed:
refactor: AOP 만 남음

leo0842 committed:
refactor: test 깨지는 부분 수정

seong-wooo committed:
refactor(appointment): appointment 도메인 재설계

leo0842 committed:
feat: 도메인 재설계 1차 완료

seong-wooo commented on backend/src/main/java/com/morak/back/brandnew/service/NewPollService.java:
```diff
@@ -0,0 +1,69 @@
+        // TODO: 2022/10/06 teamCode, memberId로 validate
+        NewPoll poll = request.toPoll(teamCode, memberId);
+
+        return pollRepository.save(poll).getPollInfo().getCode();
```
```suggestion
        pollRepository.save(poll);
        return poll.getCode();
```

seong-wooo replied:
이게 더 보기 편할 듯 하네요

seong-wooo commented on backend/src/main/java/com/morak/back/brandnew/service/NewPollService.java:
```diff
@@ -0,0 +1,69 @@
+        // TODO: 2022/10/06 teamCode, pollCode로 validate
+        NewPoll poll = pollRepository.findByCode(pollCode).orElseThrow();
+        return poll.getPollItems().stream()
+                .map(pollItem -> PollItemResultResponse.of(pollItem, poll.getPollInfo().getAnonymous()))
```
여기도 getter 두 번 쓰지말고 한 번만 쓰도록 변경하는 것이 좋을 것 같아요!

seong-wooo commented on backend/src/main/java/com/morak/back/brandnew/domain/NewPollItem.java:
```diff
@@ -0,0 +1,84 @@
+    @Embedded
+    private SelectMembers selectMembers;
+
+    public NewPollItem(Long id) {
+        this(id, null, new SelectMembers());
+    }
+
+    @Builder
+    public NewPollItem(String subject) {
+        this(null, subject, new SelectMembers());
+    }
+
+    private NewPollItem(Long id, String subject, SelectMembers selectMembers) {
+        this.id = id;
+        this.subject = subject;
+        this.selectMembers = selectMembers;
+    }
```
```suggestion
      @Builder
    private NewPollItem(Long id, String subject, SelectMembers selectMembers) {
        this.id = id;
        this.subject = subject;
        this.selectMembers = selectMembers;
    }
```

생성자를 하나로 만들죠!

seong-wooo commented on backend/src/main/java/com/morak/back/brandnew/domain/PollItems.java:
```diff
@@ -0,0 +1,56 @@
+        }
+    }
+
+    public void doPoll(Member member, Map<NewPollItem, String> data) {
+        for (NewPollItem pollItem : values) {
+            addOrRemove(pollItem, member, data);
+        }
+    }
+
+    private void addOrRemove(NewPollItem pollItem, Member member, Map<NewPollItem, String> data) {
+        if (data.containsKey(pollItem)) {
+            pollItem.addSelectMember(member, data.get(pollItem));
+            return;
+        }
+        pollItem.remove(member);
+    }
```
```suggestion
        public void doPoll(Member member, Map<NewPollItem, String> data) {
        Set<NewPollItem> newPollItems = data.keySet();
        values.removeAll(newPollItems);
        for (NewPollItem pollItem : values) {
            pollItem.remove(member);
        }
        for (NewPollItem newPollItem : newPollItems) {
            newPollItem.addSelectMember(member, data.get(newPollItem));
        }
        values.addAll(newPollItems);
    }
```

PollItems가 갖고 있는 NewPollItem들의 자료형이 Set으로 바뀐다면 이런 식으로도 가능할 듯 하네요

seong-wooo committed:
refactor(appointment): SystemTime Bean으로 등록

seong-wooo committed:
refactor(appointment): getter 변경

seong-wooo committed:
refactor(appointment): 패키지 정리

seong-wooo committed:
fix: merge from develop

seong-wooo committed:
refactor(appointment): 간접 참조로 변경

seong-wooo committed:
refactor(appointment): appointment count 계산 방법 변경 및 repositoryTest 수정

seong-wooo committed:
test(appointment): DateTest 구현

seong-wooo committed:
test(appointment): DatePeriodTest 구현

seong-wooo committed:
test(Core): ClosedAt 테스트 구현

seong-wooo committed:
test(Core): DescriptionTest 구현

seong-wooo committed:
test(Core): MenuStatusTest 구현

seong-wooo committed:
test(Core): TitleTest 구현

seong-wooo committed:
test(Core): TimeTest 구현

cjlee38 committed:
refactor: 가능시간에서 member를 member id로 변경

cjlee38 committed:
test: timePeriod Test 추가

cjlee38 committed:
test: 테스트 컴파일 오류 해결

leo0842 committed:
refactor: 서비스와 컨트롤러 이식

seong-wooo committed:
Merge branch 'dev' of https://github.com/woowacourse-teams/2022-mo-rak into refactor/new-domain

cjlee38 committed:
test: 임시 테스트 컴파일 에러 해결

leo0842 committed:
temp: 테스트용 커밋

seong-wooo committed:
test(appointment): 도메인 layer test 수정

RIANAEH committed:
feat(poll): 기존에 @Valid로 검사하던 예외에 대한 커스텀 에러 추가

RIANAEH committed:
test(poll): 재설계한 투표 도메인에 대한 테스트 추가

seong-wooo committed:
test(appointment): AppointmentSelectTest 추가

leo0842 committed:
temp: service validate

leo0842 committed:
Merge branch 'refactor/brand-new-poll-#416' of https://github.com/RIANAEH/2022-mo-rak into refactor/brand-new-poll-#416

leo0842 committed:
refactor: 서비스 테스트 반쯤 수행

RIANAEH committed:
test(poll): 재설계한 투표 서비스와 레포지토리에 대한 테스트 추가

leo0842 committed:
refactor: 서비스 테스트 절반 수행

seong-wooo committed:
10 / 17 여기서 부터 진행하셈 커밋 뒤로 돌리고

cjlee38 committed:
feat(appointment): BaseElement 구현

cjlee38 committed:
refactor: BaseEntity 패키지 이동

cjlee38 committed:
test: 불필요한 테스트 케이스 제거

cjlee38 committed:
fix(appointment): 테스트 실패케이스 로직 수정

leo0842 committed:
test: 새로 만든 도메인 로직 기존 테스트에 이식

leo0842 committed:
refactor: dto 패키지 application 패키지 내부로 이동

leo0842 committed:
refactor: 기존 dev merge commit

cjlee38 committed:
fix: 스케줄링 테스트 오류 수정

cjlee38 committed:
test: 실패하던 테스트 주석 해제

cjlee38 committed:
refactor: todo 주석 제거 및 테스트 추가

cjlee38 committed:
test: 약속잡기 추천 로직 테스트 주석 해제

cjlee38 committed:
style: reformat code 실행

leo0842 committed:
refactor: map 의 key column 을 적절하게 변경

leo0842 committed:
refactor: dto 에서 anonymous 변수명 수정

seong-wooo committed:
test(appointment): 약속시간 선택 테스트 위치 변경

leo0842 committed:
refactor: anonymous 값을 primitive 값으로 수정

leo0842 committed:
refactor: other member 변수를 메서드로 추출

RIANAEH commented on backend/src/main/java/com/morak/back/poll/domain/SelectMembers.java:
```diff
@@ -0,0 +1,67 @@
+            name = "select_member",
+            joinColumns = @JoinColumn(name = "poll_item_id")
+    )
+    @MapKeyJoinColumn(name = "member_id")
```
### MapKeyJoinColumn vs. MapKeyColumn

MapKeyJoinColumn
> Specifies a mapping to an entity that is a map key. The map key join column is in the collection table, join table, or table of the target entity that is used to represent the map. If no MapKeyJoinColumn annotation is specified, a single join column is assumed and the default values apply.

MapKeyColumn
> Specifies the mapping for the key column of a map whose map key is a basic type. If the name element is not specified, it defaults to the concatenation of the following: the name of the referencing relationship field or property; "_"; "KEY".

key인 Member가 Entity이기 때문에 MapKeyJoinColumn을 사용해야 이름이 적용됩니다. 
역할 도메인에서의 RoleName은 값 객체이기 때문에 MapKeyColumn을 사용했던 것이었습니다.

cjlee38 committed:
refactor: appointment와 team 검증 콜백 패턴 추가

leo0842 committed:
refactor: appointment 와 합치기

leo0842 committed:
refactor: menu 의 team code 를 String 에서 Code 로 변경

cjlee38 commented on backend/src/main/java/com/morak/back/poll/domain/Poll.java:
```diff
@@ -1,194 +1,97 @@
+        validateAllowedCount(data.size());
+        validateExistItem(data.keySet());
 
-    public void doPoll(Member member, Map<PollItem, String> mappedItemAndDescription) {
-        validateStatus();
-        validateCounts(mappedItemAndDescription.size());
-        validateNewItemsBelongsTo(mappedItemAndDescription.keySet());
+        pollItems.doPoll(member, data);
+    }
 
-        deleteMembersFromPollItems(member);
-        addMembersToPollItems(member, mappedItemAndDescription);
+    private void validateExistItem(Set<PollItem> selectItems) {
+        if (!pollItems.containsAll(selectItems)) {
+            throw new PollItemNotFoundException(
+                    CustomErrorCode.POLL_ITEM_NOT_FOUND_ERROR,
+                    id + "번 투표에 " +
+                            pollItems.getValues().stream()
+                                    .map(pollItem -> pollItem.getId().toString())
+                                    .collect(Collectors.joining(", ")) + "번 항목들은 투표할 수 없습니다.");
+        }
```
r: 이 검증 로직은 `PollItems` 객체가 맡는게 더 자연스러울 것 같아요

RIANAEH replied:
저는 PollItems에게 "선택한 투표 항목이 포함되어있니?"라고 물어보고 그 결과를 이용해 검증을 진행했기 때문에 괜찮다고 생각했어요. 그리고 이 검증을 PollItems에게 넘기면 Poll의 id까지 넘겨야할 것 같아요🥲

cjlee38 commented on backend/src/main/java/com/morak/back/poll/domain/SelectMembers.java:
```diff
@@ -0,0 +1,67 @@
+    )
+    @MapKeyJoinColumn(name = "member_id")
+    @Column(name = "description", nullable = false)
+    private Map<Member, String> values;
```
c: `PollInfo` 에서는 호스트를 Long 값으로 다루었는데, 여기서는 Member를 직접 들고 있어야 하나요? Member를 찾아서 내려주는 일은 application layer에 맡기는게 낫지 않을까요 ?

cjlee38 commented on backend/src/main/java/com/morak/back/poll/domain/SelectMembers.java:
```diff
@@ -0,0 +1,67 @@
+
+@Getter
+@Embeddable
+public class SelectMembers {
```
a: 이 부분은 전에 에덴한테도 한번 했던 이야기인데요. `SelectMembers`가 하는 일이 거의 없는 것 같아요. `PollItem`이 그저 `SelectMembers`에게 메소드를 전달하는 역할만 하고 있기만 하는데, `PollItem`이 직접 처리하면 어떨까요 ?

cjlee38 commented on backend/src/main/java/com/morak/back/poll/domain/PollInfo.java:
```diff
@@ -0,0 +1,101 @@
+        return this.allowedCount.isGreaterThan(itemCount);
+    }
+
+    public boolean isAllowedCountLessThan(int itemCount) {
+        return this.allowedCount.isGreaterThanOrEqual(itemCount);
+    }
```
r: 자기 자신만 사용하는 메소드인데 public으로 열려있네요. 아마 이 부분은 `AllowedCount`가 `PollItems` 쪽으로 이동하면 깔끔하게 해결될 것으로 보여요.

cjlee38 commented on backend/src/main/java/com/morak/back/poll/domain/PollInfo.java:
```diff
@@ -0,0 +1,101 @@
+    @Builder
+    public PollInfo(CodeGenerator codeGenerator, String title, Boolean anonymous, int allowedCount, Long teamId, Long hostId, PollStatus status,
+                    SystemDateTime closedAt) {
+        validateClosedAt(closedAt);
```
r: `validate`의 주체가 `PollInfo`여야 할까요 ? 종료 시각이 미래인지 과거인지를 검증하는 행위는 `closedAt` 객체가 판단해도 되지 않을까요 ?

cjlee38 commented on backend/src/main/java/com/morak/back/poll/domain/SystemDateTime.java:
```diff
@@ -0,0 +1,39 @@
+    @Column(name = "closed_at")
+    private LocalDateTime dateTime;
+
+    @Transient
+    private LocalDateTime now;
```
r: `beforeNow()`와 같은 메소드 때문에 `@Transient`로 들고 있는 것 같은데, `SystemDateTime`이 스스로 판단하게 하면 변수를 할당할 필요가 없을 것 같아요.

cjlee38 commented on backend/src/main/java/com/morak/back/poll/application/dto/PollResultRequest.java:
```diff
@@ -15,4 +16,8 @@ public class PollResultRequest {
     @NotNull(message = "poll Item description은 null 일 수 없습니다.")
     private String description;
+
+    public PollItem toPollItem() {
+        return new PollItem(id);
+    }
```
r: toPollItem으로 바꿔줘야할 이유가 없는 것 같아요. 현재 `addOrRemove` 메소드는 다음과 같은데요

```java
private void addOrRemove(PollItem pollItem, Member member, Map<PollItem, String> data) {
        if (data.containsKey(pollItem)) {
            ...
        }
}
```

아래와 같이 바꿨을 때 문제가 될 수 있을까요 ?

```java
private void addOrRemove(PollItem pollItem, Member member, Map<Long, String> data) {
        if (data.containsKey(pollItem.getId())) {
            ...
        }
}
```

cjlee38 commented on backend/src/main/java/com/morak/back/poll/domain/Poll.java:
```diff
@@ -1,194 +1,97 @@
-
-    public boolean isBelongedTo(Team otherTeam) {
-        return this.team.equals(otherTeam);
+    public int countSelectMembers() {
+        return pollItems.countSelectMembers();
```
c: 생각해보니 만약 프론트에서 `count` 가 필요없어지게 된다면 도메인 로직을 수정해야 하네요. 이 부분은 저희쪽도 고쳐야 할 것 같아요.

cjlee38 reviewed (changes requested):
고생하셨습니다~

도메인쪽만 빠르게 피드백 남겨보았습니다. 여유 되면 애플리케이션/테스트 쪽도 살펴볼게요 :)

RIANAEH commented on backend/src/main/java/com/morak/back/appointment/domain/DurationMinutes.java:
```diff
@@ -23,26 +23,17 @@ public class DurationMinutes {
     private static final int MAX_HOURS = 24;
     private static final int DAY_MINUTES = 1440;
```
C: 시간과 관련된 상수들을 enum으로 관리해보면 어떨까요?

cjlee38 replied:
잘 이해가 안되서 그러는데 어디서 enum을 활용할 수 있을까요 ?

RIANAEH commented on backend/src/test/java/com/morak/back/appointment/domain/RecommendationCellTest.java:
```diff
@@ -1,141 +1,143 @@
+        LocalDate today = LocalDate.now();
+        now = LocalDateTime.now();
+
+        DEFAULT_BUILDER = Appointment.builder()
+                .hostId(1L)
+                .teamCode(Code.generate((length) -> "TEAMcode"))
+                .title("스터디 회의 날짜 정하기")
+                .description("필참!!")
+                .code(Code.generate(length -> "MoraK123"))
+                .startDate(today.plusDays(1))
+                .endDate(today.plusDays(5))
+                .startTime(LocalTime.of(14, 0))
+                .endTime(LocalTime.of(18, 30))
+                .durationHours(1)
+                .durationMinutes(0)
+                .now(now)
+                .closedAt(now.plusDays(1));
```
R: 쓰고 있지 않는 것 같네요~

RIANAEH commented on backend/src/test/java/com/morak/back/appointment/ui/AppointmentControllerTest.java:
```diff
@@ -266,7 +264,6 @@ class AppointmentControllerTest extends ControllerTest {
     }
 
-    // TODO: 2022/07/27 범용적이게 변형 후 상속한 ControllerTest로 이전해야함!
     private ResultActions get(String path, String groupCode) throws Exception {
         return mockMvc.perform(RestDocumentationRequestBuilders.get(path, groupCode)
                 .header("Authorization", "bearer access.token"));
```
C: todo 왜 지우셨어요~ 하면 좋을 것 같은디!

RIANAEH reviewed (changes requested):
안녕하세요 까리😎
복잡한 로직이었을 텐데 리팩토링 잘 해주셨네요:)

아직 정확하게 다 파악한 것은 아닌 것 같아 1차 리뷰 남기고
이후에 더 살펴보겠습니다! (sooooo hard)

cjlee38 committed:
refactor(appointment): 약속 추천 메소드명 변경

cjlee38 commented on backend/src/main/java/com/morak/back/appointment/ui/dto/AppointmentCreateRequest.java:
```diff
@@ -51,10 +49,10 @@ public class AppointmentCreateRequest {
-    public Appointment toAppointment(Team team, Member member, Code code) {
+    public Appointment toAppointment(String teamCode, Long hostId, Code code, LocalDateTime now) {
         return Appointment.builder()
-                .team(team)
-                .host(member)
+                .teamCode(Code.generate((length) -> teamCode))
+                .hostId(hostId)
```
그렇네용. service 쪽에서 처리하도록 수정했습니다

cjlee38 commented on backend/src/main/java/com/morak/back/appointment/domain/Appointment.java:
```diff
@@ -1,229 +1,168 @@
-    @NotNull(message = "약속잡기 설명은 null일 수 없습니다.")
-    @Size(max = 1000, message = "약속잡기 설명의 길이는 0~1000자여야 합니다.")
-    private String description;
+    @Embedded
+    @Delegate
+    private Menu menu;
```
미리 공유하지 못했네요.`@Delegate` 어노테이션으로는 getter만 사용하기로 했습니다.

seong-wooo commented on backend/src/main/java/com/morak/back/appointment/application/AppointmentService.java:
```diff
@@ -119,39 +115,14 @@ public void selectAvailableTimes(String teamCode, Long memberId, String appointm
-                    availableTimeRequest + " 요청된 시간에 중복된 시간이 있습니다."
-            );
-        }
+        withTeamValidation(
+                appointment -> {
+                    appointment.selectAvailableTime(
+                            requests.stream().map(AvailableTimeRequest::getStart).collect(Collectors.toSet()),
+                            memberId,
+                            systemTime.now());
+                    return null;
+                }, teamCode, appointmentCode);
```
넵 반영했습니다!

cjlee38 committed:
refactor(appointment): service 로직 정리

cjlee38 committed:
chore: data.sql 수정 및 미사용 클래스 삭제

cjlee38 commented on backend/src/main/java/com/morak/back/appointment/domain/Appointment.java:
```diff
@@ -1,229 +1,168 @@
-    }
+        Set<AvailableTime> availableTimes = new HashSet<>();
+        for (LocalDateTime dateTime : localDateTimes) {
+            validateSelectTime(now, dateTime);
+            availableTimes.add(AvailableTime.builder().memberId(memberId).startDateTime(dateTime).build());
+        }
 
-    /*
-    DatePeriod를 직접 사용하면 getEndDate()의 minusDay 를 호출할 수 없습니다.
-     */
-    public boolean isAvailableDateRange(DatePeriod otherDatePeriod) {
-        return new DatePeriod(getStartDate(), getEndDate()).isAvailableRange(otherDatePeriod);
+        this.availableTimes.removeIf(availableTime -> availableTime.getMemberId().equals(memberId));
+        this.availableTimes.addAll(availableTimes);
     }
```
수정하다 머리아파서 일단 잠깐 홀드시켜놓을게요..

seong-wooo commented on backend/src/main/java/com/morak/back/poll/domain/PollItems.java:
```diff
@@ -0,0 +1,65 @@
+
+    @OneToMany(cascade = CascadeType.ALL)
+    @JoinColumn(name = "poll_id", nullable = false, updatable = false)
+    private List<PollItem> values;
```
NPE 조심!

cjlee38 committed:
test: 사용하지 않는 변수 제거 및 SimpleMockMvc 추가

seong-wooo commented on backend/src/main/java/com/morak/back/poll/domain/PollInfo.java:
```diff
@@ -0,0 +1,101 @@
+    }
+
+    public boolean isAllowedCount(int itemCount) {
+        return itemCount >= 1 && isAllowedCountLessThan(itemCount);
```
`return itemCount >= 1 && isAllowedCountLessThan(itemCount);`
이 한줄을 두 개의 조건으로 나누기 보다는 하나의 메서드에서 둘 다 확인해주는 것이 좋지 않을까요?
`return isAllowedCount(itemCount)` 이런 느낌?

cjlee38 committed:
refactor: menu 적용 및 getter 구현

RIANAEH committed:
refactor: core host mismatched error code 추가 및 적용

RIANAEH committed:
refactor: custom error code 관련 깨지는 테스트 코드 수정 및 사용하지 않는 custom error code 삭제

RIANAEH committed:
refactor(poll): poll 도메인에서 투표한 사람 수에 대한 의존성 제거

투표한 사람 수를 보여주는 것은 view에 의존적인 로직이라고 판단해 poll 도메인에서 이와 관련한 메서드를 제거하고 이를 dto로 이동시켰다.

RIANAEH committed:
refactor(appointment): appointment 도메인에서 투표한 사람 수에 대한 의존성 제거 및 관련 서비스 테스트 추가

leo0842 committed:
refactor: primitive 로 변경

RIANAEH committed:
refactor(poll): 생성자 시그니처 변경

cjlee38 committed:
refactor: 사용하지 않는 메소드 제거

RIANAEH commented on backend/src/main/java/com/morak/back/poll/domain/Poll.java:
```diff
@@ -1,194 +1,97 @@
-        pollItems.add(pollItem);
-    }
+    public void doPoll(Member member, Map<PollItem, String> data) {
+        validateState();
+        validateAllowedCount(data.size());
+        validateExistItem(data.keySet());
```
하핫 status로 변경했습니다!

RIANAEH commented on backend/src/main/java/com/morak/back/poll/domain/AllowedCount.java:
```diff
@@ -0,0 +1,39 @@
+@NoArgsConstructor
+@Getter
+@Embeddable
+public class AllowedCount {
```
동의합니다. 관련 로직도 PollItems로 이동시켰습니다.

cjlee38 committed:
refactor: expired time 생성자 변경

cjlee38 committed:
refactor: valid 어노테이션 제거

RIANAEH committed:
refactor(poll): 검증 메서드 명 변경

seong-wooo committed:
refactor(auth): OAuthId 값 객체 생성 및 테스트

cjlee38 committed:
fix: 실패하는 로직 및 테스트 수정

cjlee38 committed:
refactor: 사용하지 않는 메소드 제거

seong-wooo committed:
refactor(auth): 멤버 Name 값 객체 생성 및 테스트

RIANAEH committed:
refactor(poll): 투표 항목 이름 VO 생성 및 검증 추가 + SelectMembers 삭제

RIANAEH committed:
refactor(poll): 투표 선택 이 VO 생성 및 검증 추가

1000자로 제한하고, test schema도 변경했습니다.

RIANAEH committed:
refactor(poll): poll의 SystemDateTime 삭제

RIANAEH committed:
Merge branch 'refactor/brand-new-poll-#416' of https://github.com/leo0842/2022-mo-rak into refactor/brand-new-poll-#416

seong-wooo committed:
refactor(auth): 멤버 ProfileUrl 값 객체 생성 및 테스트

seong-wooo committed:
refactor(auth): 멤버 Name, OAuthId 예외 메시지 수정

seong-wooo committed:
chore: confilct 해결

leo0842 committed:
refactor: appointment 의 description 을 sub title 로 변경

cjlee38 committed:
feat(team): team name 값객체 추가

cjlee38 committed:
feat: team member 검증 로직 분리

cjlee38 committed:
test: controller 컨텍스트 캐싱

cjlee38 committed:
test: service 컨텍스트 캐싱 환경 구성

cjlee38 committed:
test: service test 캐싱 및 테스트 코드 수정

seong-wooo committed:
refactor(poll): PollService에 authorizationService 적용 및 SelectMember의 key를 Long으로 변경

leo0842 committed:
refactor: role service 와 team service 콜백 방식으로 변경

RIANAEH committed:
test: AuthorizationService 테스트 추가

leo0842 committed:
refactor: dev 와 충돌 해결

seong-wooo committed:
test: appointment 테스트 케이스 추가

RIANAEH committed:
test: Poll 테스트 추가

seong-wooo committed:
refactor(appointment): subTitle의 길이 제한 변경

seong-wooo committed:
fix(appointment): appointment_available_time 테이블의 id 제거

seong-wooo committed:
refactor(core): menu 패키지를 core로 이동

leo0842 committed:
refactor: auth 도메인 equals 메서드 test coverage 에서 제외

cjlee38 committed:
test: add tests for team domain

cjlee38 committed:
Merge branch 'refactor/brand-new-poll-#416' of https://github.com/leo0842/2022-mo-rak into refactor/brand-new-poll-#416

seong-wooo reviewed (approved):
다들 진짜 고생했어요

cjlee38 reviewed (approved):
몹 프로그래밍으로 개발 진행하였습니다.
